homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -1 +1,204 @@
 """The islamic_prayer_times component."""
+from datetime import timedelta
+import logging
+
+from prayer_times_calculator import PrayerTimesCalculator, exceptions
+from requests.exceptions import ConnectionError as ConnError
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import track_point_in_time
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    CALC_METHODS,
+    CONF_CALC_METHOD,
+    DATA_UPDATED,
+    DEFAULT_CALC_METHOD,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: {
+            vol.Optional(CONF_CALC_METHOD, default=DEFAULT_CALC_METHOD): vol.In(
+                CALC_METHODS
+            ),
+        }
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Import the Islamic Prayer component from config."""
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Islamic Prayer Component."""
+    client = IslamicPrayerClient(hass, config_entry)
+
+    if not await client.async_setup():
+        return False
+
+    hass.data.setdefault(DOMAIN, client)
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload Islamic Prayer entry from config_entry."""
+    if hass.data[DOMAIN].event_unsub:
+        await hass.async_add_executor_job(hass.data[DOMAIN].event_unsub)
+    await hass.config_entries.async_forward_entry_unload(config_entry, "sensor")
+    hass.data.pop(DOMAIN)
+
+    return True
+
+
+class IslamicPrayerClient:
+    """Islamic Prayer Client Object."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the Islamic Prayer client."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.prayer_times_info = {}
+        self.available = True
+        self.event_unsub = None
+
+    @property
+    def calc_method(self):
+        """Return the calculation method."""
+        return self.config_entry.options[CONF_CALC_METHOD]
+
+    def get_new_prayer_times(self):
+        """Fetch prayer times for today."""
+        calc = PrayerTimesCalculator(
+            latitude=self.hass.config.latitude,
+            longitude=self.hass.config.longitude,
+            calculation_method=self.calc_method,
+            date=str(dt_util.now().date()),
+        )
+        return calc.fetch_prayer_times()
+
+    def schedule_future_update(self):
```

Comment:
Change back to async context like it was before. Just schedule the data fetching on the executor. All the scheduling should remain in async context.